charts/openfga/templates/_helpers.tpl

```diff
@@ -80,3 +80,47 @@ Return true if a secret object should be created
     {{- true -}}
 {{- end -}}
 {{- end -}}
+
+
+{{- define "openfga.datastore.envConfig" -}}
+{{- if .Values.datastore.engine }}
+- name: OPENFGA_DATASTORE_ENGINE
+  value: "{{ .Values.datastore.engine }}"
+{{- end }}
+{{- if .Values.datastore.uriSecret }}
+- name: OPENFGA_DATASTORE_URI
+  valueFrom:
+    secretKeyRef:
+      name: "{{ .Values.datastore.uriSecret }}"
+      key: uri
+{{- else if and (.Values.datastore.existingSecret) (.Values.datastore.secretKeys.uriKey) }}
+- name: OPENFGA_DATASTORE_URI
+  valueFrom:
+    secretKeyRef:
+      name: "{{ .Values.datastore.existingSecret }}"
+      key: "{{ .Values.datastore.secretKeys.uriKey }}"
+{{- else if .Values.datastore.uri }}
+- name: OPENFGA_DATASTORE_URI
+  value: "{{ .Values.datastore.uri }}"
+{{- end }}
+{{- if and (.Values.datastore.existingSecret) (.Values.datastore.secretKeys.usernameKey) }}
+- name: OPENFGA_DATASTORE_USERNAME
+  valueFrom:
+    secretKeyRef:
+      name: "{{ .Values.datastore.existingSecret }}"
+      key: "{{ .Values.datastore.secretKeys.usernameKey }}"
+{{- else if .Values.datastore.username }}
+- name: OPENFGA_DATASTORE_USERNAME
+  value: "{{ .Values.datastore.username }}"
+{{- end }}
+{{- if and (.Values.datastore.existingSecret) (.Values.datastore.secretKeys.passwordKey) }}
+- name: OPENFGA_DATASTORE_PASSWORD
```

Comment:
Is there a reasoning for adding those two variables (_OPENFGA_DATASTORE_PASSWORD_ / _OPENFGA_DATASTORE_USERNAME_) directly in the env? 

I love the idea of passing them through the secret, but having the option to add them in env directly is a security risk, sensitive values should not be added as clear environment variables in the manifest in my opinion.

Reply:
How do you mean passing them through env? The secrets are also passed through an environment variable from the pod standpoint now. They are not mounted as a file or anything.

I can leave it out if wanted and just keep the old uri method for compatibility. But I think it's nice to have the option when you are doing a quick local setup.

Reply:
Yes makes sense! Come to think of it, in other charts we actually generate a secret containing the user and password as part of the chart when no existing secret is used. That way we always load from secrets.

Could follow the same principle here.